utils/test/e2e/shared_steps.py

```diff
@@ -29,6 +29,14 @@
 
 @given("the ADF pipeline {pipeline_name} has been triggered with {parameters}")
 def trigger_adf_pipeline(context, pipeline_name: str, parameters: str):
+    """
+    Trigger an Azure Data Factory pipeline with the provided parameters.
+
+    Args:
+        context (behave.runner.Context): Behave context object.
```

Comment:
Same as in the other place - please use type hints in the function signature instead of types in docstrings - it is easier to work with them in IDE.

Reply:
done